tests/network/vmi_istio.go

```diff
@@ -0,0 +1,310 @@
+		})
+		AfterEach(func() {
+			libnet.RemoveLabelFromNamespace(tests.NamespaceTestDefault, istioInjectNamespaceLabel)
+			Expect(virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.Name, &metav1.DeleteOptions{})).ShouldNot(HaveOccurred())
+			Expect(virtClient.CoreV1().Services(tests.NamespaceTestDefault).Delete(context.TODO(), service.Name, metav1.DeleteOptions{})).ShouldNot(HaveOccurred())
+		})
+		Describe("Inbound traffic", func() {
+			var job *batchv1.Job
+			AfterEach(func() {
+				var backgroundPropagation = metav1.DeletePropagationBackground
+				Expect(virtClient.BatchV1().Jobs(tests.NamespaceTestDefault).Delete(context.TODO(), job.Name, metav1.DeleteOptions{PropagationPolicy: &backgroundPropagation})).ShouldNot(HaveOccurred())
+			})
+			table.DescribeTable("request to VMI", func(port int) {
+				job, err = verifyClientServerConnectivity(vmi, port)
+
+				By("Waiting for the job to succeed")
+				Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).To(Succeed())
+			},
+				table.Entry("should reach VMI HTTP server on service declared port", svcDeclaredTestPort),
+				table.Entry("should reach VMI HTTP server on service undeclared port", svcUndeclaredTestPort),
+			)
+			Context("With PeerAuthentication enforcing mTLS", func() {
+				var strictPeerAuthentication *istiogosecv1beta1.PeerAuthentication
+				BeforeEach(func() {
+					strictPeerAuthentication = &istiogosecv1beta1.PeerAuthentication{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "default",
+							Namespace: tests.NamespaceTestDefault,
+						},
+						Spec: istiosecv1beta1.PeerAuthentication{
+							Mtls: &istiosecv1beta1.PeerAuthentication_MutualTLS{
+								Mode: istiosecv1beta1.PeerAuthentication_MutualTLS_STRICT,
+							},
+						},
+					}
+
+					virtClient, err = kubecli.GetKubevirtClient()
+					tests.PanicOnError(err)
+
+					_, err = virtClient.IstioClient().SecurityV1beta1().PeerAuthentications(tests.NamespaceTestDefault).Create(context.TODO(), strictPeerAuthentication, metav1.CreateOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+				})
+				AfterEach(func() {
+					Expect(virtClient.IstioClient().SecurityV1beta1().PeerAuthentications(tests.NamespaceTestDefault).Delete(context.TODO(), strictPeerAuthentication.Name, metav1.DeleteOptions{})).ShouldNot(HaveOccurred())
+				})
+				table.DescribeTable("request to VMI", func(port int) {
+					job, err = verifyClientServerConnectivity(vmi, port)
+
+					By("Waiting for the job to fail")
+					Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).NotTo(Succeed())
+				},
+					table.Entry("client outside mesh should not reach VMI HTTP server on service declared port", svcDeclaredTestPort),
+					table.Entry("client outside mesh should not reach VMI HTTP server on service undeclared port", svcUndeclaredTestPort),
+				)
+			})
+		})
+		Describe("Outbound traffic", func() {
+			var curlKubevirtioCommand = "curl -sD - -o /dev/null http://kubevirt.io | head -n 1\n"
```

Comment:
I'm not sure we can assume we have external connectivity.
@EdDev I remember you wanted to send a PR adding a flag allowing testing external connectivity. Was there any progress with that?

@rhrazdil is there another way to test the outbound traffic without assuming the cluster has external connectivity.

Reply:
External connectivity is supposed to be treated in this PR: https://github.com/kubevirt/kubevirt/pull/5169
It treats it by allowing the executor to mention if this is supported on a specific setup.

We probably need to do something beyond that even, by providing a ref point we setup ourself.

Reply:
@AlonaKaplan Yes, I believe I can do that.
I didn't know that we should avoid external, note that kubevirt.io is used in vmi_network.go as well.

Reply:
Done, I'm using another VMI that is not part of the mesh and acts as an external entity